spec/prog/vm/update_ipv6_spec.rb

```diff
@@ -3,118 +3,163 @@
 require_relative "../../model/spec_helper"
 
 RSpec.describe Prog::Vm::UpdateIpv6 do
-  subject(:pr) {
-    described_class.new(Strand.new)
+  subject(:pr) { described_class.new(strand) }
+
+  let(:project) { Project.create(name: "test-project") }
+  let(:location_id) { Location::HETZNER_FSN1_ID }
+
+  let(:private_subnet) {
+    PrivateSubnet.create(
+      name: "test-subnet", project_id: project.id, location_id: location_id,
+      net4: "1.0.0.0/8", net6: "fd10:9b0b:6b4b:8fbb::/64"
+    )
+  }
+
+  let(:vm_host) {
+    create_vm_host(
+      location_id: location_id,
+      total_cpus: 48, total_cores: 48, total_dies: 1, total_sockets: 1
+    )
+  }
+
+  let(:sshable) { create_mock_sshable(host: "1.1.1.1") }
+
+  let(:vm_strand) {
+    Prog::Vm::Nexus.assemble_with_sshable(
+      project.id, name: "test", private_subnet_id: private_subnet.id,
+      location_id: location_id, force_host_id: vm_host.id
+    )
   }
 
   let(:vm) {
-    instance_double(Vm,
-      inhost_name: "test",
-      storage_secrets: "storage_secrets",
-      nics: [instance_double(Nic,
-        private_subnet: instance_double(
-          PrivateSubnet,
-          net4: NetAddr::IPv4Net.parse("1.0.0.0/8")
-        ),
-        ubid_to_tap_name: "ubid_to_tap_name")],
-      name: "test",
-      vm_host_id: 1)
+    v = vm_strand.subject
+    v.update(vm_host_id: vm_host.id)
+    v
   }
 
-  let(:vm_host) {
-    instance_double(VmHost, sshable: create_mock_sshable(host: "1.1.1.1"), ip6_random_vm_network: NetAddr::IPv6Net.parse("2001:0::"))
+  let(:strand) {
+    # The VM nexus creates a strand, but we need a separate strand for UpdateIpv6
+    # that still references the same VM. Use a child strand approach.
+    Strand.create(
+      prog: "Vm::UpdateIpv6", label: "start",
+      parent_id: vm_strand.id,
+      stack: [
+        {"subject_id" => vm.id, "gpu_count" => 0, "hugepages" => true, "ch_version" => nil,
+         "gpu_device" => nil, "hypervisor" => nil, "force_host_id" => nil, "swap_size_bytes" => nil,
+         "exclude_host_ids" => [], "firmware_version" => nil, "alternative_families" => [],
+         "last_label_changed_at" => Time.now.to_s, "distinct_storage_devices" => true}
+      ]
+    )
   }
 
-  before do
-    allow(pr).to receive(:vm).and_return(vm)
+  def stub_vm_host_for_ssh
+    # Stub the prog's vm_host method to return a vm_host with mock sshable
+    stubbed_vh = VmHost[vm_host.id]
+    allow(stubbed_vh).to receive_messages(sshable: sshable, ip6_random_vm_network: NetAddr::IPv6Net.parse("2001::/64"))
```

Comment:
Why mock `sshable: sshable` here? We can create a sshable for it?

Reply:
That'll do

<img width="879" height="786" alt="Image" src="https://github.com/user-attachments/assets/6f6cea1b-b31c-43f8-baac-3a2d0b031936" />